pkg/extension/opampextension/opamp_agent_test.go

```diff
@@ -27,40 +27,98 @@ import (
-	assert.Nil(t, o.agentDescription)
+	return cfg, set
 }
 
-func TestNewOpampAgentAttributes(t *testing.T) {
-	cfg := createDefaultConfig()
-	set := extensiontest.NewNopSettings()
-	set.BuildInfo = component.BuildInfo{Version: "test version", Command: "otelcoltest"}
-	set.Resource.Attributes().PutStr(semconv.AttributeServiceName, "otelcol-sumo")
-	set.Resource.Attributes().PutStr(semconv.AttributeServiceVersion, "sumo.0")
-	set.Resource.Attributes().PutStr(semconv.AttributeServiceInstanceID, "f8999bc1-4c9b-4619-9bae-7f009d2411ec")
-	o, err := newOpampAgent(cfg.(*Config), set.Logger, set.BuildInfo, set.Resource)
-	assert.NoError(t, err)
-	assert.Equal(t, "otelcol-sumo", o.agentType)
-	assert.Equal(t, "sumo.0", o.agentVersion)
-	assert.Equal(t, "7RK6DW2K4V8RCSQBKZ02EJ84FC", o.instanceId.String())
+func setupWithRemoteConfig(t *testing.T, d string) (*Config, extension.Settings) {
+	cfg, set := defaultSetup()
+	cfg.RemoteConfigurationDirectory = d
+	return cfg, set
+}
+
+func setupWithBuildInfo(version, command string) (*Config, extension.Settings) {
+	cfg, set := defaultSetup()
+	set.BuildInfo = component.BuildInfo{Version: version, Command: command}
+	return cfg, set
+}
+
+func TestApplyRemoteConfig(t *testing.T) {
+	tests := []struct {
+		name          string
+		file          string
+		acceptsRemote bool
+		expectChanged bool
+		expectError   bool
+		errorMessage  string
+	}{
+		{"ApplyRemoteConfig", "testdata/opamp.d/opamp-remote-config.yaml", false, false, true, errMsgRemoteConfigNotAccepted},
+		{"ApplyRemoteApacheConfig", "testdata/opamp.d/opamp-apache-config.yaml", false, false, true, errMsgRemoteConfigNotAccepted},
+		{"ApplyRemoteHostConfig", "testdata/opamp.d/opamp-host-config.yaml", false, false, true, errMsgRemoteConfigNotAccepted},
+		{"ApplyRemoteWindowsEventConfig", "testdata/opamp.d/opamp-windows-event-config.yaml", false, false, true, errMsgRemoteConfigNotAccepted},
+		{"ApplyRemoteExtensionsConfig", "testdata/opamp.d/opamp-extensions-config.yaml", false, false, true, errMsgRemoteConfigNotAccepted},
+		{"ApplyRemoteConfigFailed", "testdata/opamp.d/opamp-invalid-remote-config.yaml", true, false, true, "'max_elapsed_time' must be non-negative"},
+		{"ApplyRemoteConfigMissingProcessor", "testdata/opamp.d/opamp-missing-processor.yaml", true, false, true, errMsgInvalidConfigName},
+		{"ApplyFilterProcessorConfig", "testdata/opamp.d/opamp-filter-processor.yaml", false, false, true, errMsgRemoteConfigNotAccepted},
+		{"ApplyKafkaMetricsConfig", "testdata/opamp.d/opamp-kafkametrics-config.yaml", false, false, true, errMsgRemoteConfigNotAccepted},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			d, err := os.MkdirTemp("", "opamp.d")
+			assert.NoError(t, err)
+			defer os.RemoveAll(d)
+
+			cfg := createDefaultConfig().(*Config)
+			cfg.RemoteConfigurationDirectory = d
+			set := extensiontest.NewNopSettings()
+			o, err := newOpampAgent(cfg, set.Logger, set.BuildInfo, set.Resource)
```

Comment:
might be super nit, but 2 small things:

1. this line seems to be repeated in a bunch of tests, should this be refactored as well?
2. i see that this test uses ```createDefaultConfig()```, a bunch of other tests also used that before but is now replaced with ```defaultSetup()```, should this also use that?

Reply:
done!

Reply:
```newOpampAgent``` is repeated in about 5+ tests

Reply:
addressed